recipe/post-link.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+echo "\nFor Linux 64, Open MPI is built with CUDA awareness but this support is disabled by default." >> $PREFIX/.messages.txt
```

Comment:
I think `echo` needs `-e` to convert `\n` to a proper newline (as opposed to a literal in the output). Though IIRC this is something `bash` (maybe other shells too) support and `sh` won't respect it. An easy way to solve this across shells is to break `echo`s onto multiple lines as you have done in other cases here.

ref: https://unix.stackexchange.com/a/219274

Reply:
Right, I forgot. Thanks, John!

But, do you know why even with several `echo`s, the post-link message is still printed as a one-liner? See 
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=126303&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7&l=21371
It doesn't seem to behave as expected...

Reply:
@leofang Use `printf` instead of `echo` for easy and correct newline printing.

About the bad output, It looks like a conda issue, it is likely executing a Python `print` on a byte string (note the `b'...`) prefix.

Reply:
I fixed the output. I agree it's a bug on conda/conda-build. I'm afraid no matter what we do here, we won't get a nicely formatted post-link message in the near future. Let's move on with this PR.